File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.rb
# @param {String} s
# @return {Integer}
def length_of_longest_substring(s)
  last = {}
  res = start = 0
  s.chars.each_with_index do |c, i|
    if last[c] && last[c] >= start
      start = last[c] + 1
    end
    res = [res, i - start + 1].max
    last[c] = i
  end
  res
end

File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.php
class Solution {
    /**
     * @param String $s
     * @return Integer
     */
    function lengthOfLongestSubstring($s) {
        $last = [];
        $res = $start = 0;
        $chars = str_split($s);
        foreach ($chars as $i => $c) {
            if (isset($last[$c]) && $last[$c] >= $start) {
                $start = $last[$c] + 1;
            }
            $res = max($res, $i - $start + 1);
            $last[$c] = $i;
        }
        return $res;
    }
}